pkg/operator/ceph/operator.go

```diff
@@ -100,15 +101,36 @@ func (o *Operator) Run() error {
 		return fmt.Errorf("Error starting device discovery daemonset: %v", err)
 	}
 
-	signalChan := make(chan os.Signal, 1)
-	stopChan := make(chan struct{})
-	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
-
 	serverVersion, err := o.context.Clientset.Discovery().ServerVersion()
 	if err != nil {
 		return fmt.Errorf("Error getting server version: %v", err)
 	}
 
+	if serverVersion.Major >= csi.KubeMinMajor && serverVersion.Minor >= csi.KubeMinMinor && csi.CSIEnabled() {
+		logger.Infof("Ceph CSI driver is enabled, validate csi param")
+		if err = csi.ValidateCSIParam(); err != nil {
+			logger.Warningf("invalid csi params: %v", err)
+			if csi.ExitOnError {
+				return err
+			}
+		} else {
+			csi.SetCSINamespace(namespace)
+			if err = csi.StartCSIDrivers(namespace, o.context.Clientset); err != nil {
+				csi.CSICleanup(namespace, o.context.Clientset)
```

Comment:
In the normal flow we wouldn't expect this to fail, right? If we clean up the resources it will be harder to troubleshoot. How about if we don't clean them up automatically? The next time the operator runs it should be able to try again and start the missing resources. The resources already running shouldn't be a problem since the intention was to start them.

Reply:
+1

Reply:
done